### PR TITLE
Fix guest modal and and updated test

### PR DESCRIPTION
**Pull request by Caleb-Cohen**
# Description

Allows guests to click users events, hides authors of events, provides rejection modal if guest tries to add events.

Updated testing to make rejections modals expected results.

## Type of change

Please select everything applicable. Please, do not delete any lines.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change requires an update to testing

## Issue
- [ ] Is this related to a specific issue? If so, please specify: #279 

# Checklist:

- [x] This PR is up to date with the development branch, and merge conflicts have been resolved
- [x] I have executed `npm run test-frontend` and all tests have passed successfully or I have included details within my PR on the failure
- [x] I have executed `npm run lint` and resolved any outstanding errors. Most issues can be solved by executing `npm run format`
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: client/src/features/home/CalendarPage.js [hunk 1]
```diff
@@ -8,11 +8,12 @@ import { useAuthContext } from "contexts/AuthContext";
 import { useFormModalContext } from "contexts/FormModalContext";
 import EventModal from "features/modal/EventModal";
 import { useModalContext } from "contexts/ModalContext";
+import RejectionModal from "features/modal/RejectionModal";
 
 function CalendarPage() {
   const auth = useAuthContext();
   const date = useDate();
-  const formModal = useFormModalContext();
+  const formModal = useFormModalContext()
```
Comment (intelagense) [inline client/src/features/home/CalendarPage.js:16]: Did the linter not catch this?

**Comment by intelagense** [comment]
Do you think it would be better to display something like this in the place of the missing author?
"Event author: Not available to non-logged-in users"

File: client/src/features/home/CalendarPage.js (hunk 1 above)
Comment (Caleb-Cohen) [reply client/src/features/home/CalendarPage.js:16]: apparently not.

**Comment by Caleb-Cohen** [comment]
> Do you think it would be better to display something like this in the place of the missing author? "Event author: Not available to non-logged-in users"

I can get that done.

**Pull request merged**